netweaver/extract_nw_archives.sls

```diff
@@ -0,0 +1,51 @@
+{%- from "netweaver/map.jinja" import netweaver with context -%}
+{% set nw_extract_dir = netweaver.nw_extract_dir %}
+{% set swpm_extract_dir = nw_extract_dir~'/SWPM' %}
+
+setup_nw_extract_directory:
+  file.directory:
+    - name: {{ nw_extract_dir }}
+    - mode: 755
+    - makedirs: True
+
+{% if netweaver.sapcar_exe_file is defined and netweaver.swpm_sar_file is defined %}
+
+extract_installer_file:
+    sapcar.extracted:
+    - name: {{ netweaver.swpm_sar_file }}
+    - sapcar_exe: {{ netweaver.sapcar_exe_file }}
+    - output_dir: {{ nw_extract_dir }}/SWPM
+    - options: "-manifest SIGNATURE.SMF"
+
+{% endif %}
+
+{% set additional_dvd_folders = [] %}
+
+{% for dvd in netweaver.additional_dvds %}
+{% set dvd_folder = dvd.split('.')[0] %}
+{% if additional_dvd_folders.append(dvd_folder) %}{% endif %}
+
+{%- if dvd.endswith((".ZIP", ".zip", ".RAR", ".rar")) %}
+
+extract_nw_archive:
+  archive.extracted:
+    - name: {{ dvd_folder }}
```

Comment:
I don't know if we should use `dvd_folder` here.
What if the dvd path is `my_folder/51053787`, this would extract to `/sapmedia/nw/my_folder/51053787`, is this what we want? Or do we want `/sapmedia/nw/51053787`

Reply:
Thanks I updated the logic to just fetch the trailing folder name using `salt['file.basename']` instead of including the